include/TS3.h

```diff
@@ -3,27 +3,36 @@
 
 #include <iostream>
 
-#include "TDetector.h"
+#include "TGRSIDetector.h"
 #include "TS3Hit.h"
 
-class TS3 : public TDetector {
+class TS3 : public TGRSIDetector {
 	public:
 		TS3();
-		~TS3();
+		TS3(const TS3&);
+		virtual  ~TS3();
 
-		void AddFragment(TFragment*, MNEMONIC*);
-		void BuildHits() {} //no need to build any hits, everything already done in AddFragment
+		virtual void AddFragment(TFragment*, MNEMONIC*);
+		virtual void BuildHits();
 
+		TGRSIDetectorHit* GetHit(const int& idx =0);
 		TS3Hit* GetS3Hit(const int& i);  
-		Short_t GetMultiplicity() { return fS3Hits.size(); }
+		Short_t GetMultiplicity() const { return fS3Hits.size(); }
+		void PushBackHit(TGRSIDetectorHit* deshit);
 
-		TVector3 GetPosition(int front, int back);
+		static TVector3 GetPosition(int ring, int sector);
 
-      virtual void Clear(Option_t *opt = "all");		     //!
-      virtual void Print(Option_t *opt = "") const;		  //!
+		void Copy(TObject&) const;
+		TS3& operator=(const TS3&);  // 
+		virtual void Clear(Option_t *opt = "all");		     //!
+		virtual void Print(Option_t *opt = "") const;		  //!
+		
+		
 
 	private:
 		std::vector<TS3Hit> fS3Hits;
+		std::vector<TFragment*> fS3_RingFragment; //! 
+		std::vector<TFragment*> fS3_SectorFragment; //! 
```

Comment:
I most definitely agree this is two conversations. I don't mean this to
hold up the pull request as it is more detailed analysis than coding.

On Wed, Nov 4, 2015 at 9:44 PM, jsmallcombe <notifications@github.com>
wrote:

> In include/TS3.h
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584#discussion_r43970912>
> :
>
> >
> >  	private:
> >  		std::vector<TS3Hit> fS3Hits;
> > +		std::vector<TFragment*> fS3_RingFragment; //!
> > +		std::vector<TFragment*> fS3_SectorFragment; //!
>
> Yes, certainly something to do next.
>
> I thought the issue here was collecting the fragments? Which the CSM class
> also does? I think maybe 2 conversations got confused.
>
> Pull request really should be a place to discuss "Is this code good
> enough" rather than the next development. And this is the same as the
> current (working) method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584/files#r43970912>
> .
>